skimage/morphology/footprints.py

```diff
@@ -612,41 +612,127 @@ def _cross(r0, r1, dtype=np.uint8):
     return c
 
 
-def _cross_decomposition(footprint, dtype=np.uint8):
+def footprint_cross_decompose(footprint):
     """Decompose a symmetric convex footprint into cross-shaped elements.
 
     This is a decomposition of the footprint into a sequence of
-    (possibly asymmetric) cross-shaped elements. This technique was proposed in
-    [1]_ and corresponds roughly to algorithm 1 of that publication (some
-    details had to be modified to get reliable operation).
+    (possibly asymmetric) cross-shaped elements. Applying this series of
+    smaller footprints will give a result close to (or even equivalent to) the
+    dense original `footprint` but with better computational performance for
+    large footprints.
 
+    Parameters
+    ----------
+    footprint : (N, M) ndarray
+        A symmetric convex footprint as a 2-D array of 1's and 0's.
+        N and M must be of odd length.
+
+        .. warning::
+            For performance reasons, it isn't checked whether `footprint` is
+            symmetric or convex or contains only 0 and 1. So, for unsupported
+            footprints, this function may return an incorrect decomposition!
```

Comment:
I'd appreciate early feedback here. I'm not aware of a fast way to check if large footprints are convex and symmetric. So instead I tried to add performance-friendly checks where I could and added this warning.

What do you think?

Reply:
Is there a good way of leaving this choice of slow-accurate vs fast-dangerous to the user?

Reply:
I thought about using a parameter like `check_convex_symmetry : bool` that defaults to true. For large footprints, users could disable that one. 

Or we could only check smaller footprints where the performance isn't impacted too much. And if users want explicit checking behavior the can use `check_convex_symmetry`...?

Reply:
We could also just keep this and these kind of functions private and only duplicate the existing functions with the `decomposition=` parameter. It's a trade-off between an API (a) that gives users more low-level building blocks that have to be used with care and an API (b) with fewer safer building blocks that might not support all use cases of (a).

Reply:
@stefanv suggested that may we could simply check that the decomposed footprint composes again to the dense version at the end of the function. This shouldn't impact performance too much in the big picture and if it does. I'll come up with a proposal.